Compilation error in misc/packcc/packcc.c

When attempting a fresh compile on a Linux 3.10.0 system, with gcc 4.8.5, I am seeing the following error.

```
./misc/packcc/packcc.c: In function 'unescape_string':
./misc/packcc/packcc.c:414:21: error: 'for' loop initial declarations are only allowed in C99 mode
                     for (int k=0;k<4;k++) {
                     ^
./misc/packcc/packcc.c:414:21: note: use option -std=c99 or -std=gnu99 to compile your code
./misc/packcc/packcc.c:426:25: error: 'for' loop initial declarations are only allowed in C99 mode
                         for (int k=1;k<4;k++) ch = (ch << 4) | s[k];
                         ^
make[1]: *** [packcc] Error 1
make: *** [all] Error 2
```

This can be fixed with the following patch.
```diff
diff --git a/misc/packcc/packcc.c b/misc/packcc/packcc.c
index f61274d..ffbbc58 100644
--- a/misc/packcc/packcc.c
+++ b/misc/packcc/packcc.c
@@ -411,7 +411,8 @@ static bool unescape_string(char *str, size_t *length) {
                 }
                 {
                     char s[4];
-                    for (int k=0;k<4;k++) {
+                    int k;
+                    for (k=0;k<4;k++) {
                         char c = str[i + k + 1];
                         s[k] = (c >= '0' && c <= '9') ? c - '0' :
                                (c >= 'a' && c <= 'f') ? c - 'a' + 10 :
@@ -423,7 +424,8 @@ static bool unescape_string(char *str, size_t *length) {
                     }
                     else {
                         int ch=s[0];
-                        for (int k=1;k<4;k++) ch = (ch << 4) | s[k];
+                        int k;
+                        for (k=1;k<4;k++) ch = (ch << 4) | s[k];
                         if (ch < 0x80) {
                             str[j++] = (char)ch;
                         } else if (ch < 0x800) {
```

Requesting someone implement this fix to allow compilation on more systems.

--- a/circle.yml
+++ b/circle.yml
@@ -8,10 +8,10 @@ jobs:
   #
   # We assume GNU Make is available.
   #
-   fedora32_gmake:
+   fedora33_gmake:
      working_directory: ~/universal-ctags
      docker:
-       - image: docker.io/fedora:32
+       - image: docker.io/fedora:33
      steps:
        - run:
            name: Install Git
@@ -210,5 +210,5 @@ workflows:
       - fedora30_bmake
       - fedora_distcheck
       - centos_make
-      - fedora32_gmake
+      - fedora33_gmake
       - fedora33_cross_aarch64

--- a/circle.yml
+++ b/circle.yml
@@ -121,6 +121,30 @@ jobs:
            name: Test
            command: |
              make check roundtrip CIRCLECI=1
+   centos7_make:
+     working_directory: ~/universal-ctags
+     docker:
+       - image: docker.io/centos:7
+     steps:
+       - run:
+           name: Install Git
+           command: |
+             yum -y install git
+       - checkout
+       - run:
+           name: Install build tools
+           command: |
+              yum -y gcc automake autoconf pkgconfig make libxml2-devel jansson-devel libyaml-devel findutils diffutils sudo
+       - run:
+           name: Build
+           command: |
+             bash ./autogen.sh
+             ./configure --enable-debugging
+             make -j 2
+       - run:
+           name: Test
+           command: |
+             make check roundtrip CIRCLECI=1
    ubuntu20_32bit:
      working_directory: ~/universal-ctags
      docker:

--- a/m4/ax_prog_cc_for_build.m4
+++ b/m4/ax_prog_cc_for_build.m4
@@ -37,6 +37,7 @@
 AU_ALIAS([AC_PROG_CC_FOR_BUILD], [AX_PROG_CC_FOR_BUILD])
 AC_DEFUN([AX_PROG_CC_FOR_BUILD], [dnl
 AC_REQUIRE([AC_PROG_CC])dnl
+AC_REQUIRE([AC_PROG_CC_C99])dnl
 AC_REQUIRE([AC_PROG_CPP])dnl
 AC_REQUIRE([AC_CANONICAL_BUILD])dnl
 
@@ -44,6 +45,7 @@ dnl Use the standard macros, but make them use other variable names
 dnl
 pushdef([ac_cv_prog_CPP], ac_cv_build_prog_CPP)dnl
 pushdef([ac_cv_prog_cc_c89], ac_cv_build_prog_cc_c89)dnl
+pushdef([ac_cv_prog_cc_c99], ac_cv_build_prog_cc_c99)dnl
 pushdef([ac_cv_prog_gcc], ac_cv_build_prog_gcc)dnl
 pushdef([ac_cv_prog_cc_works], ac_cv_build_prog_cc_works)dnl
 pushdef([ac_cv_prog_cc_cross], ac_cv_build_prog_cc_cross)dnl
@@ -84,6 +86,7 @@ AS_IF([test -n "$build"],      [ac_build_tool_prefix="$build-"],
 
 AC_LANG_PUSH([C])
 AC_PROG_CC
+AC_PROG_CC_C99
 _AC_COMPILER_EXEEXT
 _AC_COMPILER_OBJEXT
 AC_PROG_CPP
@@ -120,6 +123,7 @@ popdef([ac_cv_c_compiler_gnu])dnl
 popdef([ac_cv_prog_cc_g])dnl
 popdef([ac_cv_prog_cc_cross])dnl
 popdef([ac_cv_prog_cc_works])dnl
+popdef([ac_cv_prog_cc_c99])dnl
 popdef([ac_cv_prog_cc_c89])dnl
 popdef([ac_cv_prog_gcc])dnl
 popdef([ac_cv_prog_CPP])dnl

